Billets précédents et suivants : uniquement ceux choisis par le Staff

atman0eirb opened the pull request:
Fix #5793

### Contrôle qualité

  - Lancez `make zmd-start` puis `make run`.
  - Lancez `python3 manage.py test zds.tutorialv2.tests.tests_views.tests_published --settings zds.settings.test` pour le test.
  
  - Connectez-vous sur le site local avec un user normal et créez un billet.
  - Reconnectez-vous avec l'utilisateur staff et validez le billet précédent.
  - Vérifiez que le billet est mis en avant dans la page d'accueil.
  
<!-- Merci ! -->

atman0eirb committed:
Billets précédents et suivants : uniquement ceux choisis par le Staff

philippemilink commented:
Je viens de jeter un oeil au test qui ne passe pas, et tu peux le corriger ainsi : 
```py
def test_opinion_link_is_not_related_to_the_author(self):
        """
        Test that the next and previous link in the opinion page take all the opinions
        into accounts and not only the ones of the author.
        """
        opinion_not_picked = PublishedContentFactory(author_list=[self.user_author], type="OPINION")
        opinion_not_picked.save()

        user_1_opinion_1 = PublishedContentFactory(author_list=[self.user_author], type="OPINION")
        user_1_opinion_1.sha_picked = user_1_opinion_1.sha_public
        user_1_opinion_1.save()

        opinion_not_picked = PublishedContentFactory(author_list=[self.user_author], type="OPINION")
        opinion_not_picked.save()

        user_2_opinion_1 = PublishedContentFactory(author_list=[self.user_guest], type="OPINION")
        user_2_opinion_1.sha_picked = user_2_opinion_1.sha_public
        user_2_opinion_1.save()

        opinion_not_picked = PublishedContentFactory(author_list=[self.user_author], type="OPINION")
        opinion_not_picked.save()

        user_1_opinion_2 = PublishedContentFactory(author_list=[self.user_author], type="OPINION")
        user_1_opinion_2.sha_picked = user_1_opinion_2.sha_public
        user_1_opinion_2.save()

        opinion_not_picked = PublishedContentFactory(author_list=[self.user_author], type="OPINION")
        opinion_not_picked.save()

        PickListOperation.objects.create(
            content=user_1_opinion_1,
            operation="PICK",
            staff_user=self.user_staff,
            operation_date=datetime.datetime.now(),
            version=user_1_opinion_1.sha_public,
        )
        PickListOperation.objects.create(
            content=user_2_opinion_1,
            operation="PICK",
            staff_user=self.user_staff,
            operation_date=datetime.datetime.now(),
            version=user_2_opinion_1.sha_public,
        )
        PickListOperation.objects.create(
            content=user_1_opinion_2,
            operation="PICK",
            staff_user=self.user_staff,
            operation_date=datetime.datetime.now(),
            version=user_1_opinion_2.sha_public,
        )

        result = self.client.get(
            reverse("opinion:view", kwargs={"pk": user_1_opinion_2.pk, "slug": user_1_opinion_2.slug})
        )
        self.assertEqual(result.context["previous_content"].pk, user_2_opinion_1.public_version.pk)
        self.assertIsNone(result.context["next_content"])

        result = self.client.get(
            reverse("opinion:view", kwargs={"pk": user_2_opinion_1.pk, "slug": user_2_opinion_1.slug})
        )
        self.assertEqual(result.context["previous_content"].pk, user_1_opinion_1.public_version.pk)
        self.assertEqual(result.context["next_content"].pk, user_1_opinion_2.public_version.pk)
```
Je pense qu'il te manquait les lignes `user_1_opinion_2.sha_picked = user_1_opinion_2.sha_public` dans tes tentatives de corriger le test. J'en ai aussi profité pour ajouter des billets qui ne sont pas sélectionnés, de cette façon ça teste aussi ta modification.

Je te laisse faire le commit.

atman0eirb committed:
modify test  to accept the Fix #5793

modify test : test_opinion_link_is_not_related_to_the_author in
 file : zds/tutorialv2/tests/tests_views/tests_published.py 
 to accept the Fix #5793

atman0eirb commented:
Effectivement, il manquait aussi de modifier cette ligne `staff_user=self.user_author`. Merci

atman0eirb committed:
import de PickListOperation pour le test

philippemilink commented:
Je viens de lire les instructions de QA, et j'apporte quelques précisions : 
- pas la peine de lancer le test, la CI le fait pour nous
- il faut aussi (et surtout) s'assurer que les changements que proposent cette PR fonctionnent bien, à savoir : 
  - créer un billet A, le mettre en avant
  - créer un billet B
  - créer un billet C, le mettre en avant
  - créer un billet D
  - créer un billet E, le mettre en avant
  - quand on consulte le billet C, tout en bas, les billets précédent et suivant proposés doivent être respectivement A et E, pas B et D.

victorlohezic reviewed (approved):
J'ai appliqué la procédure et les billets A  et E sont bien proposés et non pas B, C, D.

philippemilink reviewed (approved):
**QA OK** ✔️

philippemilink committed:
Merge branch 'dev' into dev
